### PR TITLE
75 improve the behavior in the plugin

### LINKED ISSUE #75: Ctrl+Z will make editor scroll to top [Firefox]
### Precheck

- [x] Do a quick search and make sure the bug has not been reported yet.
- [x] Please disclose security vulnerabilities privately at [oss@kommit.co](mailto:oss@kommit.co).
- [x] Finally, be nice and have fun!.

### Description

If an article is very long, and scrollbar is in the middle of the article, press Ctrl+Z will make editor scroll to top.

this experience is not good

### Environment

* Editor.js version: 2.22.2
* Plugins you use with their versions: No plugin enabled
* Device, Browser, OS: Firefox Windown10, PC

### Current behavior

Press Ctrl+Z will scroll to top

### Expected behavior

Press Ctrl+Z would not scroll to any where

### Additional context

Firefox will cause this problem, Chrome is ok

### PR DESCRIPTION
**Pull request by MarioRodriguezS**
## Description

According to previous issues, the plugin has an unexpected behavior. The main problem is that with any undo or redo action, the plugin will render all the blocks in the editor.

So, the proposal is that the plugin updates only the involved block in the undo and redo action. 
The tests were updated as well. 

Fix #75

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -142,32 +142,129 @@ export default class Undo {
...
+    }
+    return false;
+  }
+  /**
+   * return true if the block has to be deleted becuase it was skipped previously.
+   * @param {*} index is the block index in state.
+   * @param {*} compIndex is the index to compare and know if the block was inserted previously.
+   * @returns true if a block was inserted previously and it has to be deleted.
+   */
+
```
Comment (juanhurtado10) [inline src/index.js:178]: @MarioRodriguezS Could you please format this code fragment and remove this space

File: src/index.js [hunk 2]
```diff
@@ -142,32 +142,129 @@ export default class Undo {
...
+        this.stack[this.position].index = index;
+      }
+
+      if (this.blockWasDeleted(state, nextState)) {
+        this.insertDeletedBlock(state, nextState, index);
+        return;
+      }
+
+      if (this.blockWasSkipped(index, nextIndex, state, nextState) && this.position !== 0) {
+        this.editor.blocks.delete();
```
Comment (juanhurtado10) [inline src/index.js:216]: @MarioRodriguezS is it possible to use object destructuring for `this.editor`?

```js
const {blocks, caret} = this.editor
```

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]
@MarioRodriguezS looks good 🚀 !! Some minor changes were requested!

File: src/index.js (hunk 2 above)
Comment (MarioRodriguezS) [reply src/index.js:216]: Yes, it is possible, and it is a good idea. Thanks!
Could be better set blocks and caret as this.blocks and this.caret.

**Comment by MarioRodriguezS** [comment]
@juanhurtado10 The comments were solved, thanks!

**Review by juanhurtado10 — APPROVED** [review]
✨✨✨

**Pull request merged**